Salary App/Features/SalaryForm/View/SalaryFormViewController.swift

```diff
@@ -0,0 +1,336 @@
+   private let salaryTypeTextField = UITextField()
+ 
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setup()
+    }
+    
+    @objc func dismissKeyboard() {
+        view.endEditing(true)
+    }
+}
+
+extension SalaryFormViewController {
+    // MARK: Calculation
+    
+    @objc func calculateSalary() {
+        let salaryValue = Double(salaryTextField.text ?? "") ?? 0
+        let pensionValue = Double(pensionTextField.text ?? "") ?? 0
+         do {
+            try validationManager.isValid(texts: [salaryTextField.text ?? "" , pensionTextField.text ?? ""])
+            try validationManager.inValidNumber(values: [pensionTextField.text  ?? "" , salaryTextField.text ?? ""])
+            try validationManager.isValidSalary(number: salaryValue)
+            try validationManager.isValidPenContribution(number: pensionValue)
+             
+             let monthlySalary =  salaryvm.calculateMonthlySalary(salaryForm: SalaryForm(salary: salaryValue, pensionContributions: pensionValue))
+            
+             let dailySalary =  salaryvm.calculateDailySalary(salaryForm:SalaryForm(salary: salaryValue, pensionContributions: pensionValue))
+             
+             let weeklySalary = salaryvm.calculateWeeklySalary(salaryForm: SalaryForm(salary: salaryValue, pensionContributions: pensionValue))
+             
+             let hourlySalary = salaryvm.calculateHourlySalary(salaryForm: SalaryForm(salary: salaryValue, pensionContributions: pensionValue))
+
+          
+            let tax = taxCalculator.calculate(salary: salaryValue)
+            let ni = niCalculator.calculate(salary: salaryValue)
+            let studentLoan = studentLoanCalulator.calculate(salary: salaryValue)
+
+             let sheetViewController = SalaryBreakDownViewController(salaryBreakdown: SalaryBreakdown(monthlySalary: monthlySalary, dailySalary: dailySalary, hourlySalary: hourlySalary, weeklySalary: weeklySalary, tax: tax, ni: ni, studentLoan: studentLoan, originalSalary: salaryValue))
+            
+            navigationController?.pushViewController(sheetViewController, animated: true)
+            
+        } catch FormError.containsEmptyString {
+            showAlert(title: "Error", message: FormError.containsEmptyString.errorLocalDescription, actionTitle: "Ok")
+        } catch FormError.formContainsLetters {
+            showAlert(title: "Error", message: FormError.formContainsLetters.errorLocalDescription, actionTitle: "Ok")
+        }catch FormError.invalidSalary {
+            showAlert(title: "Error", message: FormError.invalidSalary.errorLocalDescription, actionTitle: "Ok")
+        } catch FormError.isValidPenContribution {
+            showAlert(title: "Error", message: FormError.isValidPenContribution.errorLocalDescription, actionTitle: "Ok")
+        }
+        catch {
+            print("Not known\(error) ")
+        }
+        
+        
+    }
+}
+
+extension UIViewController {
```

Comment:
Move this into a seperate file, look into good naming conventions for extension files